airbyte/_processors/sql/snowflakecortex.py

```diff
@@ -0,0 +1,291 @@
+                    self._pending_state_messages[f"_{state_msg.type}"].append(state_msg)
+                else:
+                    # stream_state = cast(AirbyteStreamState, state_msg.stream)
+                    # to-do: remove hardcoding below
+                    stream_name = "myteststream"  # stream_state.stream_descriptor.name
+                    self._pending_state_messages[stream_name].append(state_msg)
+
+            else:
+                # Ignore unexpected or unhandled message types:
+                # Type.LOG, Type.TRACE, Type.CONTROL, etc.
+                pass
+
+        # in base class, following reads from expected streams, hence overriding
+        for stream_name in stream_schemas:
+            self.write_stream_data(stream_name, write_strategy=write_strategy)
+
+        # Clean up files, if requested.
+        if self.cache.cleanup:
+            self.cleanup_all()
+
+    @overrides
+    def _get_stream_config(
+        self,
+        stream_name: str,
+    ) -> ConfiguredAirbyteStream:
+        """Overriding so we can use local catalog instead of catalog manager"""
+        if not self._catalog:
+            raise exc.PyAirbyteInternalError(
+                message="Catalog should exist but does not.",
+            )
+
+        matching_streams: list[ConfiguredAirbyteStream] = [
+            stream for stream in self._catalog.streams if stream.stream.name == stream_name
+        ]
+        if not matching_streams:
+            raise exc.AirbyteStreamNotFoundError(
+                stream_name=stream_name,
+                context={
+                    "available_streams": [stream.stream.name for stream in self._catalog.streams],
+                },
+            )
+
+        if len(matching_streams) > 1:
+            raise exc.PyAirbyteInternalError(
+                message="Multiple streams found with same name.",
+                context={
+                    "stream_name": stream_name,
+                },
+            )
+
+        return matching_streams[0]
+
+    def _get_column_list_from_table(
+        self,
+        table_name: str,
+    ) -> list[str]:
+        """Get column names for passed stream."""
+        conn: Connection = self.cache.get_database_connection_via_alternate_method()
+        cursor = conn.cursor()
+        cursor.execute(f"DESCRIBE TABLE {table_name};")
+        results = cursor.fetchall()
+        column_names = [row[0].lower() for row in results]
+        cursor.close()
+        conn.close()
+        return column_names
+
+    @overrides
+    def _ensure_compatible_table_schema(
+        self,
+        stream_name: str,
+        *,
+        raise_on_error: bool = True,
+    ) -> bool:
+        """Read the exsting table schema using Snowflake python connector"""
+        json_schema = self.get_stream_json_schema(stream_name)
+        stream_column_names: list[str] = json_schema["properties"].keys()
+        table_column_names: list[str] = self._get_column_list_from_table(stream_name)
```

Comment:
get_column_list_from_table uses a local Snowflake Python connector based implementation. SQLAlchemy does not recognize vector type, hence fetching scheme (with embeddings) fails.